pkg/kubelet/kubelet_pods.go

```diff
@@ -1341,6 +1341,13 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 	spec := &pod.Spec
 	allStatus := append(append([]v1.ContainerStatus{}, s.ContainerStatuses...), s.InitContainerStatuses...)
 	s.Phase = getPhase(spec, allStatus)
+	// When restartPolicy is never and sandbox is exited, we should set pod phase to Failed
+	if len(podStatus.SandboxStatuses) != 0  && len(podStatus.ContainerStatuses) == 0 && pod.Spec.RestartPolicy == v1.RestartPolicyNever {
```

Comment:
I think the answer to this is yes, but just confirming that the first two equality statements will be true if and only if the sandbox is exited?

Reply:
do you have any good idea?